driver-bitfury16.c

```diff
@@ -0,0 +1,4589 @@
+
+				workd_list_push(info->stale_work_list, WORKD(wdata));
+				workd_list_remove(info->work_list, bitfury, &info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].cwork);
+
+				gen_task_data(info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].cwork.payload.midstate,
+						info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].cwork.payload.m7,
+						info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].cwork.payload.ntime,
+						info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].cwork.payload.nbits, mask,
+						info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].cwork.task);
+
+				ret = cmd_buffer_push(cmd_buffer,
+						info->chipboard[board_id].bcm250[bcm250_id].channel_depth, chip_address, chip_address,
+						info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].cwork, 0,
+						CHIP_CMD_TASK_WRITE, 79,
+						info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].cwork.task);
+
+				if (ret < 0)
+					applog(LOG_ERR, "%s: chipworker_thr: chip [%d:%d:%2d], error prepare send_task_cmd",
+							bitfury->drv->name,
+							board_id, bcm250_id, chip_id);
+			}
+#if 0
+			else
+				applog(LOG_ERR, "%s: chipworker_thr: chip [%d:%d:%2d], error prepare send_task_cmd: no works available",
+						bitfury->drv->name,
+						board_id, bcm250_id, chip_id);
+#endif
+			L_UNLOCK(info->stale_work_list);
+			L_UNLOCK(info->work_list);
+		}
+
+		break;
+
+	/* fill chip buffer with check status task */
+	case TASK_SENT:
+		gettimeofday(&curr_time, NULL);
+
+		if (((renonce_chip(chip_address) == 0) ||
+			(opt_bf16_renonce == RENONCE_DISABLED)) &&
+			(timediff_us(info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].status_time, curr_time) > CHIP_TASK_STATUS_INTERVAL)) {
+			applog(LOG_DEBUG, "%s: chipworker_thr: chip [%d:%d:%2d], prepare task_status_cmd",
+					bitfury->drv->name,
+					board_id, bcm250_id, chip_id);
+
+			ret = cmd_buffer_push(cmd_buffer,
+					info->chipboard[board_id].bcm250[bcm250_id].channel_depth, chip_address, chip_address,
+					work, 0, CHIP_CMD_TASK_STATUS, 0, NULL);
+
+			info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].status_time.tv_sec  = curr_time.tv_sec;
+			info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].status_time.tv_usec = curr_time.tv_usec;
+
+			if (ret < 0)
+				applog(LOG_ERR, "%s: chipworker_thr: chip [%d:%d:%2d], error prepare task_status_cmd",
+						bitfury->drv->name,
+						board_id, bcm250_id, chip_id);
+		}
+		break;
+
+	/* fill chip buffer with read nonces task */
+	case TASK_SWITCHED:
+		if ((renonce_chip(chip_address) == 0) ||
+			(opt_bf16_renonce == RENONCE_DISABLED)) {
+			applog(LOG_DEBUG, "%s: chipworker_thr: chip [%d:%d:%2d], prepare read_nonce_cmd",
+					bitfury->drv->name,
+					board_id, bcm250_id, chip_id);
+
+			ret = cmd_buffer_push(cmd_buffer,
+					info->chipboard[board_id].bcm250[bcm250_id].channel_depth, chip_address, chip_address,
+					work, 0, CHIP_CMD_READ_NONCE, 0, NULL);
+
+			if (ret < 0)
```

Comment:
driver-bitfury16.c:1637:12: warning: comparison is always false due to limited range of data type [-Wtype-limits]
    if (ret < 0)